R/combined_seasonal_output.R

```diff
@@ -0,0 +1,156 @@
+#'   - 'converged': Logical. Was the IWLS judged to have converged?
+#'
+#' burden_output:
+#'   - 'season': The season that burden levels are calculated for.
+#'   - 'high_conf_level': (only for intensity_level method) The conf_level chosen for the high level.
+#'   - 'conf_levels': (only for peak_level method) The conf_levels chosen to fit the "low", "medium", "high" levels.
+#'   - 'values': A named vector with values for "very low", "low", "medium", "high" levels.
+#'   - 'par': The fit parameters for the chosen family.
+#'       - par_1:
+#'          - For 'weibull': Shape parameter (k).
+#'          - For 'lnorm': Mean of the log-transformed observations.
+#'          - For 'exp': Rate parameter (λ).
+#'       - 'par_2':
+#'          - For 'weibull': Scale parameter (λ).
+#'          - For 'lnorm': Standard deviation of the log-transformed observations.
+#'          - For 'exp': Not applicable (set to NA).
+#'   - 'obj_value': The value of the objective function - (negative log-likelihood), which represent the minimized
+#'                  objective function value from the optimisation. Smaller value equals better optimisation.
+#'   - 'converged': Logical. TRUE if the optimisation converged.
+#'   - 'family_quant': The distribution family used for the optimization.
+#'      - 'weibull': Uses the Weibull distribution for fitting.
+#'      - 'lnorm': Uses the Log-normal distribution for fitting.
+#'      - 'exp': Uses the Exponential distribution for fitting.
+#'      - 'disease_threshold': The input disease threshold, which is also the very low level.
+#'
+#' @export
+#'
+#' @examples
+#' # Generate random flu season
+#' generate_flu_season <- function(start = 1, end = 1000) {
+#'   random_increasing_obs <- round(sort(runif(24, min = start, max = end)))
+#'   random_decreasing_obs <- round(rev(random_increasing_obs))
+#'
+#'   # Generate peak numbers
+#'   add_to_max <- c(50, 100, 200, 100)
+#'   peak <- add_to_max + max(random_increasing_obs)
+#'
+#'   # Combine into a single observations sequence
+#'   observations <- c(random_increasing_obs, peak, random_decreasing_obs)
+#'
+#'  return(observations)
+#' }
+#'
+#' season_1 <- generate_flu_season()
+#' season_2 <- generate_flu_season()
+#'
+#' start_date <- as.Date("2022-05-29")
+#' end_date <- as.Date("2024-05-20")
+#'
+#' weekly_dates <- seq.Date(from = start_date,
+#'                          to = end_date,
+#'                          by = "week")
+#'
+#' tsd_data <- tsd(
+#'   observation = c(season_1, season_2),
+#'   time = as.Date(weekly_dates),
+#'   time_interval = "week"
+#' )
+#'
+#' # Run the main function
+#' combined_data <- combined_seasonal_output(tsd_data)
+#' # Print seasonal onset results
+#' print(combined_data$onset_output)
+#' # Print burden level results
+#' print(combined_data$burden_output)
+combined_seasonal_output <- function(
+  tsd,
+  k = 5,
+  level = 0.95,
+  disease_threshold = 20,
+  family = c(
+    "poisson",
+    "quasipoisson"
+  ),
```

Comment:
Continued from above:
```suggestion
  family = c("poisson", "quasipoisson"),
  family_quant = c("weibull","lnorm", "exp"),
```